libs/filer/workspace_files_extensions_client.go

```diff
@@ -245,6 +245,17 @@ func (w *workspaceFilesExtensionsClient) Write(ctx context.Context, name string,
 
 // Try to read the file as a regular file. If the file is not found, try to read it as a notebook.
 func (w *workspaceFilesExtensionsClient) Read(ctx context.Context, name string) (io.ReadCloser, error) {
+	// Ensure that the file / notebook exists. We do this check here to avoid reading
+	// the content of a notebook called `foo` when the user actually wanted
+	// to read the content of a file called `foo`.
+	//
+	// To read the content of a notebook called `foo` in the workspace the user
+	// should use the name with the extension included like `foo.ipynb` or `foo.sql`.
+	_, err := w.Stat(ctx, name)
```

Comment:
Note: In the read-only mode for the filer, this does not introduce an additional network API call since the `stat` API calls are cached and likely prefetched.

Reply:
Thanks.

I notice that the implementation of the underlying filer (the regular workspace files one) also runs a stat on read. It does this to determine that the path is not a directory and that that triggers other behavior. This could be optimized out as well (not needed for this PR).

Reply:
Yeah, noticed that as well. Switching over to `use_qualified_names` should take care of that since we'll no longer have the extensions filer at that point.